src/inference/hmc.jl

```diff
@@ -164,52 +235,50 @@ end
 
 NUTS(args...; kwargs...) = NUTS{ADBackend()}(args...; kwargs...)
 function NUTS{AD}(
-    n_iters::Int, 
-    n_adapts::Int, 
-    δ::Float64, 
-    max_depth::Int, 
-    Δ_max::Float64, 
-    init_ϵ::Float64, 
-    ::Type{metricT}, 
+    n_adapts::Int,
+    δ::Float64,
+    max_depth::Int,
+    Δ_max::Float64,
+    init_ϵ::Float64,
+    ::Type{metricT},
     space::Tuple
 ) where {AD, metricT}
-    return NUTS{AD, space, metricT}(n_iters, n_adapts, δ, max_depth, Δ_max, init_ϵ)
+    return NUTS{AD, space, metricT}(n_adapts, δ, max_depth, Δ_max, init_ϵ)
 end
 
 function NUTS{AD}(
-    n_iters::Int,
     n_adapts::Int,
     δ::Float64,
     ::Tuple{};
     kwargs...
 ) where AD
-    NUTS{AD}(n_iters, n_adapts, δ; kwargs...)
+    NUTS{AD}(n_adapts, δ; kwargs...)
 end
 
 function NUTS{AD}(
-    n_iters::Int,
     n_adapts::Int,
     δ::Float64,
     space::Symbol...;
-    max_depth::Int=10,
+    max_depth::Int=5,
     Δ_max::Float64=1000.0,
-    init_ϵ::Float64=0.0,
-    metricT=AHMC.DiagEuclideanMetric
+    init_ϵ::Float64=0.1,
+    metricT=AHMC.DenseEuclideanMetric
```

Comment:
Let's stick to `max_depth=10` (as we are fast enough to use such large max depth now), `init_ϵ=0.0` to trigger auto step size search and `DiagEuclideanMetric` which matches Stan's default.

Reply:
Cool, can do.

Reply:
When I set this to use `init_ϵ=0.0`, `metricT=AHMC.DiagEuclideanMetric`, and `max_depth=10` it becomes unbearably slow (2.5 minutes for 6000 samples on `gdemo`). I set `max_depth` back down again to five, but I'm not sure why this should cause such a large performance regression.

Reply:
Yeah, something weird is up somewhere.